bokehjs/src/lib/core/ui_events.ts

```diff
@@ -634,7 +634,8 @@ export class UIEventBus {
     } else {
       let result = false
       for (const tool of this._tools.keys()) {
-        result ||= emit(tool)
+        const emitted = emit(tool)
+        result ||= emitted
```

Comment:
worth a line comment explaining that this split needs to be preserved in order to prevent short-circuiting